### PR TITLE
embedding operator manifests with go-bindata

### PR DESCRIPTION
**Pull request by jasmingacic**
So this code will embed operator-manifests as binary into CLI

To embed it i ran this 
```bash
# ../../bin/kustomize edit set image ghcr.io/calyptia/core-operator:v1.0.10
# $GOPATH/go-bindata -prefix "../core-operator" -o operator-manifest/manifest_data.go -pkg=operatormanifest ../core-operator/config/...
```
this created operator-manifest/manifest_data.go file which has all the binary data to deploy the operator
then i passed this to kubectl apply -k /tmp/config/default which spun up the operator

Requirements:
* https://github.com/go-bindata/go-bindata
* kustomize

### CODE REVIEW

File: cmd/operator/install.go [hunk 1]
```diff
@@ -110,34 +112,29 @@ func NewCmdInstall() *cobra.Command {
...
+			return "", nil
+		}
 
-	_, err = temp.WriteString(withImage)
-	if err != nil {
-		return "", err
 	}
-
-	return temp.Name(), err
+	return filepath.Join(tmpdir, "config", "default"), nil
```
Comment (gabrielbussolo) [inline cmd/operator/install.go:137]: here we recreate the yaml's based on this manifest_data.go?

**Review by gabrielbussolo — COMMENTED** [review]
I did not understood very well how it works...

from where the yamls will come and which type are they? is that one compiled "all in one", or the divided one (as on operator project)?
why do we transform it on go code and (as i understand) we bring it back to yaml?
how can we manipulate it to create or not the namespace and insert different images ?
how we manage version of the manifests?

sorry im a bit lost

File: main.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	cmd "github.com/calyptia/cli/cmd"
 )
 
+//go:generate go-bindata -modtime 1 -prefix "../../" -o operator-manifest/manifest_data.go -pkg=cmd -ignore=debug/ -ignore=local/ -ignore=prometheus/ -ignore=samples/ ../../config/... manifests/...
```
[the remark this answers is not in the document]
Comment (jasmingacic) [reply main.go:12]: The same thing is done with go:embed with the difference that go:embed cannot embed directories.

**Comment by gabrielbussolo** [comment]
Screenshot 2023-07-28 at 10 46 36
which files would be needed to embed on the cli to be using it with '-k' ?

**Comment by jasmingacic** [comment]
Files in: 
- ../default
- ../crd
- ../rbac
- ../manager

**Pull request closed without merging**